src/utils/global/tooltip/Tooltip.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { useRef, useState } from 'react'
+
+import { classes } from '../../classes'
+import styles from './tooltip.module.scss'
+
+interface TooltipProps {
+  text: string;
+  ariaLabel: string;
+  children: React.ReactNode;
+  className?: string;
+  fontSize?: string;
+  textContainerWidth?: string;
+}
+
+const Tooltip: React.FC<TooltipProps> = (props) => {
+  const [show, setShow] = useState(false);
+
+  const containerRef = useRef<HTMLDivElement>(null);
+  const tooltipRef = useRef<HTMLDivElement>(null);
+
+  // use ref and calculate position on render to determine where to place tooltip
+  const containerRect = containerRef.current?.getBoundingClientRect();
+  const tooltipRect = tooltipRef.current?.getBoundingClientRect();
+
+  const containerCenter =
+    (containerRect?.top ?? 0) + (containerRect?.height ?? 0) / 2;
+
+  // set tooltip style based on incoming props
+  const tooltipStyle = {
+    opacity: show ? 1 : 0,
+    // restrict width of tooltip with optional prop
+    width: props.textContainerWidth ? props.textContainerWidth : "auto",
+    fontSize: props.fontSize ? props.fontSize : "1rem",
+  };
+
```

Comment:
```suggestion

  // add event listeners to the children to show/hide tooltip
  const childrenWithProps = React.Children.map(props.children, (child) => {
    if (React.isValidElement(child)) {
      return React.cloneElement(child as React.ReactElement, {
        onMouseEnter: () => setShow(true),
        onMouseLeave: () => setShow(false),
      });
    }
    return child;
  });

```
To avoid having to wrap the children in two additional layers, we can add the event bindings directly to the children